### PR TITLE
Update all-for-one.md

### PR DESCRIPTION
**Pull request by gagliardetto**
This PR is a draft of how I would address the major points of confusion for bug bounty application submitters, and remove obstacles that might discourage and create friction for anyone considering to write and submit a query.

Preview: https://github.com/gagliardetto/securitylab/blob/patch-2/.github/ISSUE_TEMPLATE/all-for-one.md

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/all-for-one.md [hunk 1]
```diff
@@ -1,32 +1,87 @@
...
+- [ ] I will provide the result(s) **privately** to the Security Lab.
+
+**OR**
 
-*List the CVE ID(s) associated with this vulnerability. GitHub will automatically link CVE IDs to the [GitHub Advisory Database](https://github.com/advisories).*
+- [ ] The vulnerability is already **fixed and disclosed**.
+	- Description: URL to vulnerable code
 
-- CVE-20nn-nnnnn
+## 5. CVE ID(s)
```
Comment (xcorail) [inline .github/ISSUE_TEMPLATE/all-for-one.md:74]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/all-for-one.md [hunk 2]
```diff
@@ -1,32 +1,87 @@
...
 
-## Report
+### Instructions ❓
```
Comment (xcorail) [inline .github/ISSUE_TEMPLATE/all-for-one.md:76]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/all-for-one.md [hunk 3]
```diff
@@ -1,32 +1,87 @@
...
 
-*Describe the vulnerability. Provide any information you think will help GitHub assess the impact your query has on the open source community.*
+List the CVE ID(s) associated with this vulnerability. GitHub will automatically link CVE IDs to the [GitHub Advisory Database](https://github.com/advisories).
```
Comment (xcorail) [inline .github/ISSUE_TEMPLATE/all-for-one.md:78]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/all-for-one.md [hunk 4]
```diff
@@ -1,32 +1,87 @@
...
 
-- [ ] Are you planning to discuss this vulnerability submission publicly? (Blog Post, social networks, etc). *We would love to have you spread the word about the good work you are doing*
 
-## Result(s)
+### Your answer 👇
```
Comment (xcorail) [inline .github/ISSUE_TEMPLATE/all-for-one.md:81]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/all-for-one.md [hunk 5]
```diff
@@ -1,32 +1,87 @@
...
 
-*Provide at least one useful result found by your query, on some revision of a real project.*
+- Existing CVEs that my query would have been able to find if they weren't already fixed:
+	1. CVE-20nn-nnnnn
```
Comment (xcorail) [inline .github/ISSUE_TEMPLATE/all-for-one.md:84]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/all-for-one.md [hunk 6]
```diff
@@ -1,32 +1,87 @@
...
 
-- [description](url)
+- Vulnerabilities that my query found and then resulted in a CVE:
+	1. CVE-20nn-nnnnn
```
Comment (xcorail) [inline .github/ISSUE_TEMPLATE/all-for-one.md:87]:
```suggestion
## 3. Social

### Instructions ❓

Are you planning to discuss your query publicly? (Blog Post, social networks, etc).

**We would love to [help you] spread the word about the good work you are doing.**

### Your answer 👇

- [ ] Yes
- [ ] No
- [ ] Yes, I already have: [link](link)

```

File: .github/ISSUE_TEMPLATE/all-for-one.md [hunk 7]
```diff
@@ -1,32 +1,87 @@
...
+- [ ] No
+- [ ] Yes, I already have: [link](link)
+
+## 4. Result(s)
+
+### Instructions ❓
+
+- Provide at least one useful result found by your query, on some revision of a real project.
+- If the result(s) is **fixed and disclosed**, then you can link it in this PR.
+- If the result(s) is **NOT** fixed **nor disclosed**, then you can privately share your result via email to [security@github.com](mailto:security@github.com?subject=[BugBounty]%20Issue%20#000%20useful%20result) or on the [Security Lab slack](https://ghsecuritylab.slack.com/) sending it to `@TODO`.
```
Comment (xcorail) [inline .github/ISSUE_TEMPLATE/all-for-one.md:59]:
```suggestion
- If the result(s) is **NOT YET** fixed **nor disclosed**, and you are still waiting for a CVE, then you can privately share your result via email to [security@github.com](mailto:security@github.com?subject=[BugBounty]%20Issue%20#000%20useful%20result)
```

File: .github/ISSUE_TEMPLATE/all-for-one.md [hunk 8]
```diff
@@ -1,32 +1,87 @@
...
+- Even though your query is **out in the wild** (and it's quite trivial to run it against a sizable batch of projects), let's **not make things easy** for anyone wanting to exploit vulnerable projects.
```
Comment (xcorail) [inline .github/ISSUE_TEMPLATE/all-for-one.md:60]:
```suggestion
- Even though the vulnerabilities found by your query might be **out in the wild** (and it's quite trivial to run it against a sizable batch of projects), let's **not make things easy** for anyone wanting to exploit vulnerable projects.
```

File: .github/ISSUE_TEMPLATE/all-for-one.md [hunk 9]
```diff
@@ -1,32 +1,87 @@
...
+- We understand that contacting maintainers of all the vulnerable repositories found by your query is a hard and lenghty process, and that's why we will add a bonus if you will do that **for at least X repositories** and get a [CVE assigned for them].
```
Comment (xcorail) [inline .github/ISSUE_TEMPLATE/all-for-one.md:61]:
```suggestion
```

Comment (xcorail) [reply .github/ISSUE_TEMPLATE/all-for-one.md:61]: Still under brainstorming whether we transform The Big Slayer to do that. Let's wait until we have more clarity

File: .github/ISSUE_TEMPLATE/all-for-one.md [hunk 10]
```diff
@@ -1,32 +1,87 @@
...
+- Anyway, we're here for **automating things away** and if you want to leave the heavy lifting of finding and notifying vulnerable repositories' owners to GitHub security bots, that's fine with us.
```
Comment (xcorail) [inline .github/ISSUE_TEMPLATE/all-for-one.md:62]:
```suggestion
- We're here for **securing open source at scale** and if your query finds many vulnerable repositories, requiring a big coordinated disclosure, let's discuss and work together.
```

File: .github/ISSUE_TEMPLATE/all-for-one.md [hunk 11]
```diff
@@ -1,32 +1,87 @@
...
+
+- [ ] Yes
+- [ ] No
+- [ ] Yes, I already have: [link](link)
+
+## 4. Result(s)
+
+### Instructions ❓
+
+- Provide at least one useful result found by your query, on some revision of a real project.
```
Comment (xcorail) [inline .github/ISSUE_TEMPLATE/all-for-one.md:57]:
```suggestion
- Provide at least one useful result found by your query, on some revision of a real project, under the form of a list of CVEs.
```

File: .github/ISSUE_TEMPLATE/all-for-one.md [hunk 12]
```diff
@@ -1,32 +1,87 @@
...
+- Even though your query is **out in the wild** (and it's quite trivial to run it against a sizable batch of projects), let's **not make things easy** for anyone wanting to exploit vulnerable projects.
+- We understand that contacting maintainers of all the vulnerable repositories found by your query is a hard and lenghty process, and that's why we will add a bonus if you will do that **for at least X repositories** and get a [CVE assigned for them].
+- Anyway, we're here for **automating things away** and if you want to leave the heavy lifting of finding and notifying vulnerable repositories' owners to GitHub security bots, that's fine with us.
+- But in any case, we need proof that you **did your own reaserch** on [real projects], and succeeded in finding at least one **true positive result [through your query]**, proving that is it a **real vulnerability** that happens in real apps (and not a baseless assumption).
```
Comment (xcorail) [inline .github/ISSUE_TEMPLATE/all-for-one.md:63]:
```suggestion
- If for some reason you cannot provide a CVE, but if you can provide evidence that your query found at least one **real security vulnerability** and if you are confident that the pattern can be found in open source projects, please provide these elements.
```

File: .github/ISSUE_TEMPLATE/all-for-one.md [hunk 13]
```diff
@@ -1,32 +1,87 @@
...
+
+1. What is the vulnerability?
+	- Answer: ...
+1. How does the vulnerability work?
+	- Answer: ...
+1. What strategy do you use in your query to find the vulnerability?
+	- Answer: ...
+1. How have you reduced the number of **false positives**?
+	- Answer: ...	
+1. Etc.
```
Comment (xcorail) [inline .github/ISSUE_TEMPLATE/all-for-one.md:36]:
```suggestion
1. Other information?
```

File: .github/ISSUE_TEMPLATE/all-for-one.md [hunk 14]
```diff
@@ -1,32 +1,87 @@
...
+
+### Your answer 👇 (select one)
+
+- [ ] I will provide the result(s) **privately** to the Security Lab.
+
+**OR**
 
-*List the CVE ID(s) associated with this vulnerability. GitHub will automatically link CVE IDs to the [GitHub Advisory Database](https://github.com/advisories).*
+- [ ] The vulnerability is already **fixed and disclosed**.
+	- Description: URL to vulnerable code
```
Comment (xcorail) [inline .github/ISSUE_TEMPLATE/all-for-one.md:72]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/all-for-one.md [hunk 15]
```diff
@@ -1,32 +1,87 @@
...
+- Even though your query is **out in the wild** (and it's quite trivial to run it against a sizable batch of projects), let's **not make things easy** for anyone wanting to exploit vulnerable projects.
+- We understand that contacting maintainers of all the vulnerable repositories found by your query is a hard and lenghty process, and that's why we will add a bonus if you will do that **for at least X repositories** and get a [CVE assigned for them].
+- Anyway, we're here for **automating things away** and if you want to leave the heavy lifting of finding and notifying vulnerable repositories' owners to GitHub security bots, that's fine with us.
+- But in any case, we need proof that you **did your own reaserch** on [real projects], and succeeded in finding at least one **true positive result [through your query]**, proving that is it a **real vulnerability** that happens in real apps (and not a baseless assumption).
+
+### Your answer 👇 (select one)
+
+- [ ] I will provide the result(s) **privately** to the Security Lab.
+
+**OR**
```
Comment (xcorail) [inline .github/ISSUE_TEMPLATE/all-for-one.md:69]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/all-for-one.md [hunk 16]
```diff
@@ -1,32 +1,87 @@
...
+- Even though your query is **out in the wild** (and it's quite trivial to run it against a sizable batch of projects), let's **not make things easy** for anyone wanting to exploit vulnerable projects.
+- We understand that contacting maintainers of all the vulnerable repositories found by your query is a hard and lenghty process, and that's why we will add a bonus if you will do that **for at least X repositories** and get a [CVE assigned for them].
+- Anyway, we're here for **automating things away** and if you want to leave the heavy lifting of finding and notifying vulnerable repositories' owners to GitHub security bots, that's fine with us.
+- But in any case, we need proof that you **did your own reaserch** on [real projects], and succeeded in finding at least one **true positive result [through your query]**, proving that is it a **real vulnerability** that happens in real apps (and not a baseless assumption).
+
+### Your answer 👇 (select one)
```
Comment (xcorail) [inline .github/ISSUE_TEMPLATE/all-for-one.md:65]:
```suggestion
### Your answer 👇 
```

File: .github/ISSUE_TEMPLATE/all-for-one.md [hunk 17]
```diff
@@ -1,32 +1,87 @@
...
+
```
Comment (xcorail) [inline .github/ISSUE_TEMPLATE/all-for-one.md:66]:
```suggestion
- Existing CVEs that my query would have been able to find if they weren't already fixed:
	1. CVE-20nn-nnnnn

- Vulnerabilities that my query found and then resulted in a CVE:
	1. CVE-20nn-nnnnn	

  **OR**
```

**Review by xcorail — CHANGES_REQUESTED** [review]
Hey @gagliardetto thanks for this PR

Sorry for all the suggestions, it's a bit messy but TL;DR is
- Move back the CVE requirement up into the Results section
- Move down the social section
- Align the Results section instructions to the CVE requirement

Let's iterate after this first pass, as all the suggestions make the PR difficult to read

**Comment by xcorail** [comment]
👋🏾 @gagliardetto 
Note that this PR will need to be closed / revisited after [this one](https://github.com/github/securitylab/pull/468), which is moving the template from classical `.md` template to an issue form

cc @pwntester

**Comment by gagliardetto** [comment]
Thank you @xcorail !

Sorry, I completely forgot about this one.

**Pull request closed without merging**